### PR TITLE
TagSet: numTags and tag array as private

### PR DESCRIPTION
**Pull request by Gustl22**
`TagSet` is sometimes unsafe and not always clear at which position the `numTags` counter is pointing. This should be more convenient.

### CODE REVIEW

File: vtm/src/org/oscim/tiling/source/mapfile/MapDatabase.java [hunk 1]
```diff
@@ -978,8 +971,7 @@ private boolean processWays(QueryParameters queryParameters, ITileDataSink mapDa
                     GeoPoint[][] wayNodesArray = wayNodes.toArray(new GeoPoint[wayNodes.size()][]);
                     if (!filterRequired || !wayFilterEnabled || wayFilterBbox.intersectsArea(wayNodesArray)) {
                         List<Tag> tags = new ArrayList<>();
-                        for (int i = 0; i < e.tags.numTags; i++)
-                            tags.add(e.tags.tags[i]);
+                        tags.addAll(Arrays.asList(e.tags.getTags()));
```
Comment (devemux86) [inline vtm/src/org/oscim/tiling/source/mapfile/MapDatabase.java]: Should use the iteration over the actual tags size, like done with POIs above:
```java
for (int i = 0; i < e.tags.size(); i++)
    tags.add(e.tags.get(i));
```

File: vtm/src/org/oscim/tiling/source/mapfile/MapDatabase.java [hunk 2]
```diff
@@ -872,11 +868,10 @@ private boolean processWays(QueryParameters queryParameters, ITileDataSink mapDa
             byte numberOfTags = (byte) (specialByte & WAY_NUMBER_OF_TAGS_BITMASK);
 
             if (numberOfTags != 0) {
-
+                // Reset to ensure experimental tileBitmask tags aren't included
+                e.tags.clear();
```
Comment (devemux86) [inline vtm/src/org/oscim/tiling/source/mapfile/MapDatabase.java]: Probably not needed, since `ReadBuffer.readTags` clears tags at start anyway?

Comment (Gustl22) [reply vtm/src/org/oscim/tiling/source/mapfile/MapDatabase.java]: Sounds good!

File: vtm/src/org/oscim/tiling/source/mapfile/MapDatabase.java (hunk 1 above)
Comment (Gustl22) [reply vtm/src/org/oscim/tiling/source/mapfile/MapDatabase.java]: Oh yes! I adopted this from an older path, where i reseted the whole list and null all values. Thanks!

**Comment by devemux86** [comment]
@Gustl22 will you make the changes or I could in the merge?

**Comment by Gustl22** [comment]
Sorry i browsed the Github help pages, but i didn't get it, how to submit changes. I did a force push. Maybe it's better to do a normal push next time. So the proposed approval changes weren't removed :/

**Comment by devemux86** [comment]
> Sorry I browsed the Github help pages, but I didn't get it, how to submit changes.

Like your 2nd commit, you can make or revert changes in your pull request with subsequent commits in your branch. They appear automatically in the pull request. And we can squash them later.
With force push need to be careful to not corrupt the pull request.

**Comment by devemux86** [comment]
Thanks!

**Comment by devemux86** [comment]
Thanks!
Squashed and merged.

**Pull request merged**